### ISSUE TITLE
Try to enable PunkBuster

### ISSUE THREAD

**Issue by bgauduch** [bgauduch/call-of-duty-2-docker-server#37]
Requested by @LSL1337, see discussion in issue #1

**Comment by bgauduch** [comment]
* On the server side : add the PB binary and config file, and update the server start command to something like `+set sv_punkbuster "1" +fs_path pb +exec pbsv.cfg`
* On the client side : PB should be installed, see here: http://www.callofdutyview.net/downloads/call-of-duty-punkbuster-downloads/

**Comment by LSL1337** [comment]
first of all, thanks for the help
I was going crazy for 2 hours, couldn't make it work.
your script only mounts the main folder
so after 2 hours, after trying everything, i just realized i have to mount the pb folder as well.

btw why did you choose this setup?

**Comment by bgauduch** [comment]
@LSL1337 you are welcome !
Sorry to hear you spend so much time on this issue 😢 

You indeed need to mount the `pb` folder as well, I did not need pb with the server so far.

This setup allow to apply basic Separation of Concern principle : the image only run the server process, the game file / config are independent of the server lifecycle (you can even hot-reload the config from the server container to pick up some changes if needed).

Moreover I cannot add the `.iwd` files in the image because it would be way too big, and will cause copyright issue.

It would be awesome if you have time to push a PR adding PB setup 😉

**Comment by bgauduch** [comment]
Allright @LSL1337, I finished the multi-version build setup using Github actions so I did spend some time on adding Punkbuster support today !

You can follow the progress on the #41 PR 😉 
It also provide a rework of the server configuration setup, feel free to provide feedback on this as well !

Everything is running fine on the server side, but I still have some trouble on the client side (using cod2 steam version) : PunkBuster local binary complain about version. Not sure how to fix it for now.
I will keep you updated on this issue #42

**Comment by bgauduch** [comment]
This issue will be auto-closed once #41 is merged.
@LSL1337 feel free to open again if anything goes wrong on your side about PB.

**Comment by LSL1337** [comment]
it's working fine.
BTW i don't use docker compose, i'm on a different OS (unraid), but managed to get it to work with my own config file.
In the current compose file, you still only mount the main folder.
I can see the PB files are part of the docker image. Why is there 2 folders? what will it do my locally mounted pb folder? is it a new parameter, on top of the ded linux files?

**Comment by bgauduch** [comment]
@LSL1337 I'm not sure to fully understand your question so feel free to ask again if it's not clear enough.

I choosed to add the PB files directly in the Dockerfile because they are library dependancies and not game files.

The expected game structure is the following:
* `/SERVER_PATH`
  * ` /main`: containing all `*.iwd` and server config files
  * ` /pb`: PunkBuster libraries

If you mount a volume when running the container it will override the existing files already present.
So in your case, mounting the `/main/pb` will override PB libraries already present in the container by your own files.

You can still use run command arguments to enable or disable PunkBuster, but it will use your mounted PB libraries. You can refer to the latest `docker-compose.yaml` file for reference.

Hope it answer your question !

**Comment by LSL1337** [comment]
i removed my PB folder mount, it works great still on my setup
in your current compose, pb folder is not mounted, and PB is set to 0 btw.
you don't need to use the dedicated 1-2 command btw.

the only thing left for me, is to get the server log to show up.
I'm running PAM mode, from the mounted zpam folder. it is using the config file from there, but the mp server log is not there
any idea?

**Comment by bgauduch** [comment]
Nice to hear @LSL1337  !

Yes there is no need to mount it so far as the libraries are included in the image and the pb config file is in the main folder.

I choosed to disable PB by default, so that it can be enabled only if needed (updated the server setup doc in #48 btw).

Yeah you might be right, to be honest I was just toying around with some settings yesterday and just leaved it like that. Do you have an explanation on this parameters ?

Can you open a new issue regarding server log file ? I'm quite sure it has something to do with the way `fs_basepath` and `fs_homepath` works by default. 
I did a little hack to redirect server game log to container stdout [here](https://github.com/bgauduch/call-of-duty-2-docker-server/blob/master/Dockerfile#L68), but I'm still not happy of the dependency it has to the parameters above.
It's time to tackle this and close #33 !
